src/Utils/diagnose_buoyancy_flux.jl

```diff
@@ -0,0 +1,27 @@
+using QuadGK
+
+"""
+    diagnose_buoyancy_flux(b, z, time_period, depth)
+
+Returns the buoyancy flux (m²/s³) required to extract all the buoyancy from a 1D buoyancy profile `b(z)`
+down to some z-coordinate `depth` (< 0) over a `time_period` τ (s) by integrating the buoyancy profile
+and converting the integrated buoyancy content to an equivalent buoyancy flux:
+
+          1  ⌠ z_surface
+    Qb = --- |           [b(z) - b(depth)] dz
+          τ  ⌡ depth
+"""
+function diagnose_buoyancy_flux(b, z, time_period, depth)
+
+    Δz = z[2] - z[1]
+    z_surface = last(z)
+    @assert first(z) < depth < z_surface
+
+    ℑb = InterpolatedProfile(b, z, Δz, z_surface)
+    b₀ = ℑb(depth)
+
+    integrand(z) = ℑb(z) - b₀
+    B, err = quadgk(integrand, depth, z_surface)
```

Comment:
Shouldn't you do a finite volume integral?

This could also be implemented if we had a `SummedField` or `IntegratedField` in Oceananigans!

Reply:
I doubt `quadgk` is equivalent to a sum... ? It's less accurate despite having the appearance of being more sophisticated.

If we are using `RegularRectilinearGrid` then we just need to compute a sum over the finite volumes and multiply by the constant grid spacing to obtain an integral accurate to machine precision.

Reply:
For stretched rectilinear grids, you just sum the field mulitplied by the local vertical grid spacing.

For curvilinear grids I think a bit more care is needed; you may want to multiply by cell volume and divide the result by the integrated horizontal area (or something --- it depends what you mean by "integral").

Reply:
Hmmm not sure why horizontally curvilinear grids are relevant here since it's only acting on a 1D profile? I guess for the finite volume integral you need the horizontal area of the column?

What's happening is that a linear interpolation of the b(z) is constructed which is then integrated numerically using `quadgk`.

Numerically integrating a linear interpolation should be the same as summing over the finite volumes right? Since each cell/volume stores its mean.